### ISSUE TITLE
livereload doesn't work through ngrok

### ISSUE THREAD

**Issue by rabidaudio** [rabidaudio/swank#4]

**Comment by morgs32** [comment]
Yea what can we do about it?

**Comment by rabidaudio** [comment]
Well, the free version of `ngrok` only allows 1 port to be tunneled I believe. And we can't use the same port for the livereload server and the main server. Possibly we could spin up two `ngrok` servers, one for each?

**Comment by rabidaudio** [comment]
Actually, it seems possible to run live-reload on the same port by [using it as middleware](https://www.npmjs.com/package/tiny-lr#middleware). I'm experimenting with this

**Comment by morgs32** [comment]
But in that section it says this: "The port you listen on is important, and tinylr should always listen on the LiveReload standard one: 35729."

**Comment by rabidaudio** [comment]
That's true if you are using a browser extension for live reload. But swank uses [connect-livereload](https://github.com/intesso/connect-livereload) to inject a javascript reload script into HTML pages, so browser extensions aren't necessary. In fact, they might conflict, since to would probably cause the app to refresh twice.

However, the livereload server needs `/connect` and `/livereload.js` endpoints, so you couldn't host files that would match those.

**Comment by morgs32** [comment]
check out the script that connect-livereload injects into your html pages. it uses port 35729 by default, let's you change it, but doesn't let you remove it.

**Comment by rabidaudio** [comment]
What's wrong with setting it to port 80? That way it will be tunneled through ngrok. I guess if you're using HTTPS that might cause a problem.

Would you advocate instead running two ngrok tunnels?

**Comment by morgs32** [comment]
Actually, I'm using [angular-fullstack](https://github.com/angular-fullstack/generator-angular-fullstack), which uses connect-livereload, but no matter what I try, I can't change the port whatsoever.

**Comment by rabidaudio** [comment]
If you're using angluar-fullstack, why use swank? The grunt task has almost everything you need built in. My recommendation is to add a grunt task that starts ngrok. 

If you sign up for an api key, you might be able to tunnel multiple ports through the same URL. Then you can just pass both the app port and 35729.

**Comment by morgs32** [comment]
the problem is that the script that gets put into every html page wants to
point to ngrok.io:35729. any ideas about how to workaround that?

**Comment by rabidaudio** [comment]
Can I ask what you need both for anyway? Usually if you're using livereload you're working on your own machine

**Comment by rabidaudio** [comment]
By the way, it seems you can set [`src`](https://github.com/intesso/connect-livereload#options) in `connect-livereload` opts to change the whole url
